### PR TITLE
Remove minor compiler warnings

### PR DESCRIPTION
**Pull request by steven-bellock**
Signed-off-by: Steven Bellock <sbellock@nvidia.com>

### CODE REVIEW

**Comment by steven-bellock** [comment]
@ekohandel I think these warnings should have been caught by your compiler flag changes in #859. Not sure why they weren't reported.

File: library/spdm_crypt_lib/libspdm_crypt_crypt.c [hunk 1]
```diff
@@ -3336,10 +3308,6 @@ bool libspdm_req_asym_sign_hash(
         /* Need use SPDM 1.2 signing*/
 
         switch (req_base_asym_alg) {
-        case SPDM_ALGORITHMS_BASE_ASYM_ALGO_TPM_ALG_SM2_ECC_SM2_P256:
-            param = "";
```
Comment (jyao1) [inline library/spdm_crypt_lib/libspdm_crypt_crypt.c:3340]: @steven-bellock , have you verified the different between NULL and "" in SM2?

Comment (steven-bellock) [reply library/spdm_crypt_lib/libspdm_crypt_crypt.c:3340]: `param` would map to the `id_a` parameter in that function. But I see what you mean. Let me try this a different way.

**Comment by steven-bellock** [comment]
Will start a new pull request.

**Comment by ekohandel** [comment]
> @ekohandel I think these warnings should have been caught by your compiler flag changes in #859. Not sure why they weren't reported.

@steven-bellock if I understand correctly the change here is adding a `const` qualifier for the `context` which is a string. This is then used as a parameter to `libspdm_asym_verify_func` which has that parameter defined as a `const uint8_t *` type.

The change in #859 only generates a warning if a qualifier is removed but in this case a qualifier is added. The intention here would be to catch trying to mutate memory that is immutable but it would be ok to mark mutable memory as immutable.

**Pull request closed without merging**